tests/schema/tests.py

```diff
@@ -1545,6 +1547,63 @@ def test_check_constraints(self):
         if not any(details['columns'] == ['height'] and details['check'] for details in constraints.values()):
             self.fail("No check constraint for height found")
 
+    @skipUnlessDBFeature('supports_column_check_constraints')
+    def test_remove_field_check_does_not_remove_custom_checks(self):
+        # Create table
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+
+        # Add the custom check constraint
+        constraint = CheckConstraint(check=Q(height__gte=0), name='author_height_gte_0_check')
+        custom_constraint_name = constraint.name
+
+        try:
+            Author._meta.constraints = [constraint]
+            with connection.schema_editor() as editor:
+                editor.add_constraint(Author, constraint)
+
+            # Ensure the constraints exist
+            constraints = self.get_constraints(Author._meta.db_table)
+            self.assertIn(custom_constraint_name, constraints)
+            self.assertEqual(
+                len([name
+                     for name, details in constraints.items()
+                     if details['columns'] == ['height'] and details['check'] and name != custom_constraint_name]),
+                1)
+
+            # Alter the column to remove field check
+            old_field = Author._meta.get_field('height')
+            new_field = IntegerField(null=True, blank=True)
+            new_field.set_attributes_from_name('height')
+            with connection.schema_editor() as editor:
+                editor.alter_field(Author, old_field, new_field, strict=True)
+            constraints = self.get_constraints(Author._meta.db_table)
+            self.assertIn(custom_constraint_name, constraints)
+            self.assertEqual(
+                len([name
+                     for name, details in constraints.items()
+                     if details['columns'] == ['height'] and details['check'] and name != custom_constraint_name]),
+                0)
+
+            # Alter the column to re-add field check
+            new_field2 = Author._meta.get_field('height')
+            with connection.schema_editor() as editor:
+                editor.alter_field(Author, new_field, new_field2, strict=True)
+            constraints = self.get_constraints(Author._meta.db_table)
+            self.assertIn(custom_constraint_name, constraints)
+            self.assertEqual(
+                len([name
+                     for name, details in constraints.items()
+                     if details['columns'] == ['height'] and details['check'] and name != custom_constraint_name]),
+                1)
+
+            # Drop the check constraint
+            with connection.schema_editor() as editor:
+                AuthorWithUniqueName._meta.constraints = []  # sqlite hack
```

Comment:
I think we'd want more details about why this hack is needed.

Reply:
Isn't the `AuthorWithUniqueName` an error? Shouldn't it be `Author` in this test? (The similar line in the other tests use the model adjusted in that test.) 

But if so, it's not causing a failure on SQLite, so what's it for?

Reply:
added details

chaged `AuthorWithUniqueName` to `Author`, it doesn't raise because `remove_constraint` for sqlite use `model._meta.constraints` when recreate table and just had incorrect state in database after test run